app/src/org/commcare/android/adapters/EntitySubnodeDetailAdapter.java

```diff
@@ -32,10 +33,28 @@ public EntitySubnodeDetailAdapter(Context context, Detail detail, List<TreeRefer
         this.context = context;
         this.detail = detail;
         this.references = references;
-        this.entities = entities;
-        this.modifier = modifier;
+        this.initializeData(references, entities);
     }
 
+    /**
+     * Populate entities and references. Will ignore any entities (and their
+     * associated references) without any valid fields.
+     */
+    private void initializeData(List<TreeReference> references, List<Entity<TreeReference>> entities) {

```

Comment:
Can you rename the method arguments here so that they don't conflict with class fields? Doing so will enable you to omit the `this` keyword.